src/test/app/Offer_test.cpp

```diff
@@ -851,26 +858,114 @@ class Offer_test : public beast::unit_test::suite
     }
 
     void
-    testcaseSelfCross()
+    testSelfCross(bool use_partner)
     {
-        testcase ("Self-crossing");
+        testcase (std::string("Self-crossing") +
+            (use_partner ? ", with partner account" : ""));
 
         using namespace jtx;
 
         auto const gw = Account {"gateway"};
+        auto const partner = Account {"partner"};
         auto const USD = gw["USD"];
         auto const BTC = gw["BTC"];
 
         Env env {*this};
         env.fund (XRP (10000), gw);
-        env (offer (gw, BTC (500), USD (100)));
-        env (offer (gw, USD (100), BTC (500)));
-        env.close ();
+        if (use_partner)
+        {
+            env.fund (XRP (10000), partner);
+            env (trust (partner, USD (100)));
+            env (trust (partner, BTC (500)));
+            env (pay (gw, partner, USD(100)));
+            env (pay (gw, partner, BTC(500)));
+        }
+        auto const& account_to_test = use_partner ? partner : gw;
+
+		env.close();
```

Comment:
It looks like a couple of tab characters have slipped in the back door.  Personally, I've been treating the code base as using spaces only for vertical white space, and I think that's pretty consistent across our files.